### PR TITLE
Update README to introduce Quark Script only

### PR DESCRIPTION
**Pull request by haeter525**
<p align="center">
    <a href="https://www.blackhat.com/asia-21/arsenal/schedule/index.html#quark-engine-storyteller-of-android-malware-22458">
        Black Hat Arsenal
    </a>
    <a href="https://conference.hitb.org/hitb-lockdown002/sessions/quark-engine-an-obfuscation-neglect-android-malware-scoring-system/">
        HITB
    </a>
    <a href="https://www.youtube.com/watch?v=SOH4eqrv9_g&ab_channel=ROOTCONHackingConference">
        rootcon
    </a>
    <a href="https://www.youtube.com/watch?v=XK-yqHPnsvc&ab_channel=DEFCONConference">
        defcon
    </a><br>
    <a href="https://github.com/quark-engine/quark-engine/actions/workflows/pytest.yml">
        build status
    </a>
    <a href="https://codecov.io/gh/quark-engine/quark-engine">
        codecov
    </a>
    <a href="https://github.com/18z/quark-rules/blob/master/LICENSE">
        license
    </a>
    <a href="https://www.python.org/downloads/release/python-360/">
        python version
    </a>
    <a href="https://pypi.org/project/quark-engine/">
        PyPi Download
    </a><br>
    <a href="https://twitter.com/quarkengine">
        Twitter
    </a><br>
    
</p>

# Quark Script - Dig Vulnerabilities in the BlackBox

### Innovative & Interactive

*   The goal of Quark Script aims to provide an innovative way for mobile security researchers to analyze or **pentest** the targets.
*   Based on Quark, we integrate decent tools as Quark Script APIs and make them exchange valuable intelligence to each other. This enables security researchers to **interact** with staged results and perform **creative** analysis with Quark Script.

### Dynamic & Static Analysis

*   In Quark script, we integrate not only static analysis tools (e.g. Quark itself) but also dynamic analysis tools (e.g. [objection](https://github.com/sensepost/objection)).

### Re-Usable & Sharable

*   Once the user creates a Quark script for specific analysis scenario. The script can be used in another targets. Also, the script can be shared to other security researchers. This enables the exchange of knowledges.

### More APIs to come

*   Quark Script is now in a beta version. We'll keep releasing practical APIs and analysis scenarios.
*   **See API document [here](https://quark-engine.readthedocs.io/en/latest/quark_script.html#introduce-of-quark-script-apis).**

# CWE Showcases

[CWE-89](https://quark-engine.readthedocs.io/en/latest/quark_script.html#detect-cwe-89-in-android-application-androgoat-apk) | [CWE-94](https://quark-engine.readthedocs.io/en/latest/quark_script.html#detect-cwe-94-in-android-application-ovaa-apk) | [CWE-312](https://quark-engine.readthedocs.io/en/latest/quark_script.html#detect-cwe-312-in-android-application-ovaa-apk) | [CWE-319](https://quark-engine.readthedocs.io/en/latest/quark_script.html#detect-cwe-319-in-android-application-ovaa-apk) | [CWE-532](https://quark-engine.readthedocs.io/en/latest/quark_script.html#detect-cwe-532-in-android-application-dvba-apk) | [CWE-749](https://quark-engine.readthedocs.io/en/latest/quark_script.html#detect-cwe-749-in-android-application-mstg-android-java-apk) | [CWE-780](https://quark-engine.readthedocs.io/en/latest/quark_script.html#detect-cwe-780-in-android-application-mstg-android-java-apk) | [CWE-798](https://quark-engine.readthedocs.io/en/latest/quark_script.html#detect-cwe-798-in-android-application-ovaa-apk) | [CWE-921](https://quark-engine.readthedocs.io/en/latest/quark_script.html#detect-cwe-921-in-android-application-ovaa-apk) | [CWE-926](https://quark-engine.readthedocs.io/en/latest/quark_script.html#detect-cwe-926-in-android-application-dvba-apk)

# Quick Start

In this section, we will show how to detect CWE-798 with Quark Script.

### Step 1: Environments Requirements

*   Quark requires Python 3.8 or above.

### Step 2: Install Quark Engine

*   Install Quark Engine by running:

```bash
$ pip3 install -U quark-engine
```

### Step 3: Prepare Quark Script, Detection Rule and the Sample File

1.  Get the CWE-798 Quark Script and the detection rule [here](https://quark-engine.readthedocs.io/en/latest/quark_script.html#detect-cwe-798-in-android-application-ovaa-apk).
2.  Get the sampe file (ovaa.apk) [here](https://github.com/dark-warlord14/ovaa/releases/tag/1.0).
3.  Put the script, detection rule, and sample file in the same directory.
4.  Edit accordingly to the file names:

```bash
SAMPLE_PATH = "ovaa.apk"
RULE_PATH = "findSecretKeySpec.json"
# Now you are ready to run the script!
```

### Step 4: Run the script

```bash
$ python3 CWE-798.py

# You should now see the detection result in the terminal.
Found hard-coded AES key 49u5gh249gh24985ghf429gh4ch8f23f
```

*   **Check the [document](https://quark-engine.readthedocs.io/en/latest/quark_script.html#quark-script) for more examples.**

# Acknowledgments

### The Honeynet Project

<a href="https://www.honeynet.org"> Honeynet.org logo </a>

### Google Summer Of Code

Quark-Engine has been participating in the GSoC under the Honeynet Project!

*   2021:
    *   [YuShiang Dang](https://twitter.com/YushianhD): [New Rule Generation Technique & Make Quark Everywhere Among Security Open Source Projects](https://quark-engine.github.io/2021/08/17/GSoC-2021-YuShiangDang/)
    *   [Sheng-Feng Lu](https://twitter.com/haeter525): [Replace the core library of Quark-Engine](https://quark-engine.github.io/2021/08/17/GSoC-2021-ShengFengLu/)

Stay tuned for the upcoming GSoC! Join the [Honeynet Slack chat](https://gsoc-slack.honeynet.org/) for more info.

# Core Values of Quark Engine Team

*   We love **battle fields**. We embrace **uncertainties**. We challenge **impossibles**. We **rethink** everything. We change the way people think. And the most important of all, we benefit ourselves by benefit others **first**.

### CODE REVIEW

**Comment by haeter525** [comment]
The PR moves all the introductions of Quark reports to the document (quark_report.rst). Here is the result.

---

# Quark Report

## Easy to Use and Reading Friendly Report

Quark provides **7 flexible report formats** to boost your analysis.

* Summary Report
* Detail Report
* Web Report
* Label-based Report
* Behaviors Comparison Radar Chart
* Call Graph
* Rule Classification

Please see below for more details.

## Summary Report

Examine with rules.

```bash
quark -a 14d9f1a92dd984d6040cc41ed06e273e.apk -s
```

There is the possibility to select only one label to filter the rules:

```bash
quark -a 14d9f1a92dd984d6040cc41ed06e273e.apk -s network
```

There is also the possibility to select only one rule:

```bash
quark -a 14d9f1a92dd984d6040cc41ed06e273e.apk -s <path_to_the_rule>
```



Note that if you want to select the default rules of Quark, the path to the ruleset is `$HOME/.quark-engine/quark-rules/rules/`.

## Detail Report

This is how we examine a real android malware (candy corn) with one single rule (crime).

```bash
quark -a 14d9f1a92dd984d6040cc41ed06e273e.apk -d
```

There is the possibility to select only one label to filter the rules:

```bash
quark -a 14d9f1a92dd984d6040cc41ed06e273e.apk -d network
```

There is also the possibility to select only one rule:

```bash
quark -a 14d9f1a92dd984d6040cc41ed06e273e.apk -d <path_to_the_rule>
```



Note that if you want to select the default rules of Quark, the path to the ruleset is `$HOME/.quark-engine/quark-rules/rules/`.

## Web Report

With the following command, you can easily analyze the Android sample and output the web report. See our demo [here](https://pulorsok.github.io/ruleviewer/web-report-demo).


```bash
quark -a sample.apk -s -w quark_report.html
```



## Label-based Report

Check which topic (indicated by [labels](https://github.com/quark-engine/quark-rules/blob/master/label_desc.csv)) of the malware is more aggressive.

```bash
quark -a Ahmyth.apk -l detailed
```



## Behaviors Comparison Radar Chart

With the following command, you can compare different APK actions based on the max confidence of rule labels and generate a radar chart.

```bash
quark -a first.apk -a second.apk -C
```



## Call Graph

You can add the `-g` option to the quark command, and you can get the call graph (only those rules match with 100% confidence).

```bash
quark -a Ahmyth.apk -s -g
```



## Rule Classification

You can add the `-c` option to the quark command, and you can output the rules classification with the mutual parent function (only those rules match with 100% confidence).

```bash
quark -a Ahmyth.apk -s -c
```

File: docs/source/quark_reports.rst [hunk 1]
```diff
@@ -0,0 +1,129 @@
+++++++++++++++++++++++++++++++++++++++++
+Quark Report
+++++++++++++++++++++++++++++++++++++++++
+
+Easy to Use and Reading Friendly Report
```
Comment (sidra-asa) [inline docs/source/quark_reports.rst:5]: How about "reader-friendly report" or "intuitive report", instead of "Reading Friendly Report"

Comment (haeter525) [reply docs/source/quark_reports.rst:5]: Thanks for pointing out the typo! Will correct it.

Comment (haeter525) [reply docs/source/quark_reports.rst]: I've change it to "Intuitive Report". Thanks.

**Comment by 18z** [comment]
LGTM, @18z approved this PR.

**Review by 18z — APPROVED** [review]

**Comment by sidra-asa** [comment]
LGTM. Prepare to merge.

**Pull request merged**